sklearn/cluster/_bisect_k_means.py

```diff
@@ -0,0 +1,729 @@
+            picked_samples = leaves_dict[parent_id]["samples"]
+
+            # Pick data and weights to bisect
+            picked_data = X[picked_samples]
+            picked_weights = sample_weight[picked_samples]
+
+            # Perform Bisection
+            _centers, _labels = self._bisect(picked_data, picked_weights, random_state)
+
+            if self.verbose:
+                print(f"Centroid Found: {_centers[0]}")
+                print(f"Centroid Found: {_centers[1]}")
+
+            if strategy == "biggest_sse":
+                # "biggest_sse" uses computed SSE (Sum of Squared Errors)
+                # to pick cluster with biggest SSE Error
+                metrics_values = self._compute_bisect_errors(
+                    picked_data, _centers, _labels, picked_weights
+                )
+            else:
+                # strategy == 'largest_cluster'
+                # 'largest_cluster' takes counts occurances of each label
+                # to pick cluster with largest number of data points assigned
+                metrics_values = np.bincount(_labels)
+
+            # "Create Hierarchy":
+            # Cluster with smaller metrics value (SSE or ammount of points)
+            # will be at 'left side' and cluster with higher at 'right side'
+            ordered_labels = (
+                (0, 1) if metrics_values[0] <= metrics_values[1] else (1, 0)
+            )
+
+            # Assign calculated nested clusters to their root
+            tree_dict[parent_id]["children"] = [last_center_id, last_center_id + 1]
+
+            for ii, label in enumerate(ordered_labels):
+                child_id = tree_dict[parent_id]["children"][ii]
+
+                # Save Results on Tree
+                tree_dict[child_id] = {"children": None, "center": _centers[label]}
+
+                # Create Mask for samples for selecting proper data points
+                samples_mask = leaves_dict[parent_id]["samples"].copy()
+                samples_mask[picked_samples] = _labels == label
+
+                # Save recently generated leaves
+                leaves_dict[child_id] = {
+                    "samples": samples_mask,
+                    "error_or_size": metrics_values[label],
+                }
+
+            # Split cluster is no longer leaf
+            del leaves_dict[parent_id]
+
+            last_center_id += 2
+
+            # Pick new 'biggest cluster' to split
+            parent_id, _ = max(leaves_dict.items(), key=lambda x: x[1]["error_or_size"])
+
+        # Delete Initial cluster
+        del tree_dict[-1]
+
+        # Initialize Labels
+        labels = np.full(X.shape[0], -1, dtype=np.intc)
+
+        centers = []
+
+        for i, (leaf_id, leaf) in enumerate(leaves_dict.items()):
+            # Save leaf cluster centers
+            centers.append(tree_dict[leaf_id]["center"])
+
+            # Assign labels to proper data points
+            labels[leaf["samples"]] = i
+
+        centers = np.asarray(centers)
```

Comment:
centers should be an array allocated from the start, and the loop should fill the appropriate rows of the array